cql3/statements/select_statement.cc

```diff
@@ -600,7 +600,7 @@ indexed_table_select_statement::prepare_command_for_base_query(query_processor&
 future<coordinator_result<std::tuple<foreign_ptr<lw_shared_ptr<query::result>>, lw_shared_ptr<query::read_command>>>>
 indexed_table_select_statement::do_execute_base_query(
         query_processor& qp,
-        dht::partition_range_vector&& partition_ranges,
+        dht::chunked_partition_range_vector&& partition_ranges,
         service::query_state& state,
```

Comment:
Why would this be a large vector? How large would it be?

Reply:
I don't see where we limit the number of partitions if there are no restrictions on the indexed column, no LIMIT, and the query is unpaged.

Reply:
I added a unit test that reproduces the large allocation

Reply:
If we don't limit it, won't we OOM?

Using chunked containers is a band-aid if there is no intrinsic limit.